config/jobs/kubernetes-sigs/kind/kind-presubmits.yaml

```diff
@@ -242,6 +242,100 @@ presubmits:
             # during the tests more like 3-20m is used
             cpu: 2000m
   # mimic pull-kubernetes-e2e-kind, but using kind built in this PR
+  - always_run: true
```

Comment:
- copy from k/k forked job
- add extra ref for the kubernetes branch
- update `name` field to match kind presubmits
- update container `command` to cd to k/k and run e2e.sh instead of curl + e2e-k8s.sh
- drop `cluster` field since it's unclear if we're allowed to use that cluster

Reply:
aha: the missing piece is to also drop the `branches` field entirely, as we want it to run on whatever the kind development branch is.

Reply:
https://github.com/kubernetes/test-infra/pull/20770